Tensile/KernelWriterAssembly.py

```diff
@@ -7345,12 +7706,31 @@ def localReadDo(self, kernel, bufferIdx, iui, epsi, tP):
           paramList.append(vgpr("LocalReadAddr%s"%tc))
           for oIdx in range(0, numOffsets):
             if kernel["MatrixInstruction"]:
-              tIdx = tP["tensorIdx"]
-              readOffsetWidth = kernel["MacroTile%u" % tIdx] // kernel["ThreadTile%u" % tIdx]
-              if tP["isB"]:
-                readOffsetWidth *= (kernel["ThreadTile1"] // kernel["MatrixInstN"]) * 4 # 4 simds
-              paramList.append(((rIdx * blockWidth * readOffsetWidth + kernel["SubGroup%u" % tIdx]*(vIdx*numOffsets+oIdx)*kernel["VectorWidth"] \
-                + tP["localReadOffset"])*tP["bpe"]+tP["localReadSwapByteOffset"])//offsetMultiplier)
+              if not kernel["TransposeLDS"]:
+                tIdx = tP["tensorIdx"]
+                readOffsetWidth = kernel["MacroTile%u" % tIdx] // kernel["ThreadTile%u" % tIdx]
+                if tP["isB"]:
+                  readOffsetWidth *= (kernel["ThreadTile1"] // kernel["MatrixInstN"]) * 4 # 4 simds
+                paramList.append(((rIdx * blockWidth * readOffsetWidth + kernel["SubGroup%u" % tIdx]*(vIdx*numOffsets+oIdx)*kernel["VectorWidth"] \
+                  + tP["localReadOffset"])*tP["bpe"]+tP["localReadSwapByteOffset"])//offsetMultiplier)
+              else:
+                #transposeLDS case
+                # offset = K iteration offset + 'block-offset' for each row / column block + ldsPadOffset
+                # per-load-write = DepthU*LSPA/4*BPE + LdsPadOffset
+                # block-offset = (per-load-write) * InstructionTileOuput
+                # ldsPadOffset = (per-load-write)/LDsBoundaryPerPAd)*LdsPad*BPE
+                #perWavePerP = (kernel["LSP%s"%tc] // ((tP["glvw"] * globalParameters["WavefrontWidth"]) // kernel["DepthU"]))
+                #perLdsWrSize =  perWavePerP * kernel["DepthU"] * tP["bpe"]
+                #TransposeLDS do not support ds_read2 opcodes
+                assert(numOffsets<2)
+                if tc == "A":
+                  InstructionTileOuput = kernel["MIWG0"]
+                else:
+                  InstructionTileOuput = (globalParameters["WavefrontWidth"] // kernel["MIWG0"] ) * (kernel["MatrixInstN"] // kernel["InstSplit"])
+                blockOffset = (InstructionTileOuput) * kernel["DepthU"] * tP["bpe"]
+                ldsPadOffset = (blockOffset//kernel["LdsBlockSizePerPad"])*kernel["LdsPad%s"%tc]*tP["bpe"]
+                offset = (blockOffset + ldsPadOffset)*rIdx + uIdx*kernel["MatrixInstK"]*tP["bpe"] + tP["localReadSwapByteOffset"]
+                paramList.append(offset)
```

Comment:
Looks the same as the block above. Can this be simplified?

Reply:
they are not same.. for 'B' output space depends upon how wg0,wg1 mapped ; works out differently for mfma instructions